graphql/mod.ts

```diff
@@ -0,0 +1,62 @@
+import { Drash } from "../deps.ts";
+import {
+  ExecutionResult,
+  graphql,
+  GraphQLSchema,
+  renderPlaygroundPage,
+} from "./deps.ts";
+
+interface GraphQLOptions {
+  schema: GraphQLSchema;
+  graphiql: boolean;
+  rootValue: Record<string, () => string>;
+}
+
+/**
+ * Taken from https://github.com/deno-libs/gql/blob/master/http.ts but heavily modified to suit drash's needs,
+ * and to utilise as much of graphql's own code instead
+ */
+export function GraphQL(
+  options: GraphQLOptions,
+): (
+  request: Drash.Http.Request,
+  response: Drash.Http.Response,
+) => Promise<Drash.Http.Response> {
+  return async (request: Drash.Http.Request, response: Drash.Http.Response) => {
+    if (
+      options.graphiql && request.method === "GET" &&
+      (request.headers.get("Accept")?.includes("text/html") ||
+        request.headers.get("Accept")?.includes("*/*"))
+    ) {
+      response.headers.set("Content-Type", "text/html");
+      response.body = renderPlaygroundPage({ endpoint: "/graphql" });
```

Comment:
if we hardcode the playground endpoint, it probably makes more sense as `/graphiql`, right? it would be awesome if we do something like this tho (taken from below):

`GraphQL({ 
    schema,
    graphiql: { endpoint:  "/some-endpoint" } },
    rootValue: root
});`

Reply:
Well actually the only reason i did `/graphql` was because that seems to be what graphql does itself, though i get graphiql does make more sense

So now i'm quite torn :/ Do we go with the more 'official' version, or go with a more semantic version?

Reply:
eg in the second code block [here](https://graphql.org/graphql-js/running-an-express-graphql-server/) (with the JS code), it shows that the playground interface is `/graphql`

Reply:
but i have updated it slightly: https://github.com/drashland/deno-drash-middleware/pull/118/commits/d6852f935adcb62adce0c46e5db853b42109c92f

Reply:
oh, i see! yeah if we are trying to do a 1:1 that makes sense.

Reply:
why the additional check `typeof options.graphiql === "string"` ?

Reply:
I guess it's just saying "if the user sets graphiql to true, use our default endopoint, else if its a string, it means they passed their custom endpoint so use that" - do you have a better impl in mind?

Reply:
```
const playgroundEndpoint = getPlaygroundEndpoint();

function getPlaygroundEndpoint() {
  if (options.graphiql === true) {
    return "/graphql";
  }
  if (typeof options.graphiql === "string") {
    return options.graphiql
  }
}
```

wouldn't mind something more verbose, but tbh ... nitpick lol it's not a big deal at all